pyEpiabm/pyEpiabm/sweep/spatial_sweep.py

```diff
@@ -113,32 +112,20 @@ def find_infectees(self, infector_cell: Cell,
         """
         infectee_list = []
         # Chooses a list of cells (with replacement) for each infection
-        # event to occur in. Specifically inter-cell infections
-        # so can't be the same cell.
+        # event to occur in.
+        # Specifically inter-cell infections so can't be the same cell.
         distance_weights = []
-        for cell2 in possible_infectee_cells:
-            try:
-                distance_weights.append(1/DistanceFunctions.dist(
-                            infector_cell.location, cell2.location))
-            except ZeroDivisionError:
-                # If cells are on top of each other use nan placeholder
-                distance_weights.append(np.nan)
-        # Cells on top of each currently have a distance weight equal
-        # to the maximum of all other weights.
-        # Possibly want to do twice this.
-        number_of_nans = sum(np.isnan(distance_weights))
-        if number_of_nans == len(distance_weights):
-            distance_weights = [1 for _ in distance_weights]
-        elif number_of_nans > 0:
-            max_weight = np.nanmax(distance_weights)
-            distance_weights = np.nan_to_num(distance_weights,
-                                             nan=max_weight)
         # Use of the cutoff distance idea from CovidSim.
         cutoff = Parameters.instance().infection_radius
-        distance_weights = [weight if (cutoff > 1/weight) else 0
-                            for weight in distance_weights]
         # Will catch the case if distance weights isn't configured
         # correctly and returns the wrong length.
+        for cell2 in possible_infectee_cells:
+            if cell2.id in infector_cell.nearest_neighbours.keys():
+                distance_weights.append(
+                    infector_cell.nearest_neighbours[cell2.id])
+            else:
+                distance_weights.append(0)
```

Comment:
Whole section re-written

Reply:
That looks better (although [filtering](https://www.programiz.com/python-programming/methods/built-in/filter) the list in place rather than regenerating it case by case will definitely still be faster!)